integration/resources/templates/combination/graphqlapi_pipeline_resolver.yaml

```diff
@@ -0,0 +1,177 @@
+Transform: AWS::Serverless-2016-10-31
+Resources:
+  MyApiKey:
+    Type: AWS::AppSync::ApiKey
+    Properties:
+      ApiId: !GetAtt SuperCoolAPI.ApiId
+
+  DynamoDBPostsTable:
+    Type: AWS::DynamoDB::Table
+    Properties:
+      TableName: posts-table
+      AttributeDefinitions:
+      - AttributeName: id
+        AttributeType: S
+      KeySchema:
+      - AttributeName: id
+        KeyType: HASH
+      ProvisionedThroughput:
+        ReadCapacityUnits: 5
+        WriteCapacityUnits: 5
+
+  DynamoDBPostsLogTable:
+    Type: AWS::DynamoDB::Table
+    Properties:
+      TableName: posts-log-table
+      AttributeDefinitions:
+      - AttributeName: id
+        AttributeType: S
+      KeySchema:
+      - AttributeName: id
+        KeyType: HASH
+      ProvisionedThroughput:
+        ReadCapacityUnits: 5
+        WriteCapacityUnits: 5
+
+  SuperCoolAPI:
+    Type: AWS::Serverless::GraphQLApi
+    Properties:
+      Name: PradsApi
+      SchemaInline: |
+        schema {
+            query: Query
+            mutation: Mutation
+        }
+
+        type Query {
+            getPost(id: String!): Post
+        }
+
+        type Mutation {
+            addPost(
+            author: String!,
+            title: String!,
+            content: String!,
+            url: String!
+            ): Post!
+        }
+
+        type Post {
+            id: String!
+            author: String
+            title: String
+            content: String
+            url: String
+            ups: Int!
+            downs: Int!
+            version: Int!
+        }
+      Auth:
+        Type: API_KEY
+      DynamoDBDataSources:
+        PostsDataSource:
+          TableName: !Ref DynamoDBPostsTable
+          TableArn: !GetAtt DynamoDBPostsTable.Arn
+        PostsLogDataSource:
+          TableName: !Ref DynamoDBPostsLogTable
+          TableArn: !GetAtt DynamoDBPostsLogTable.Arn
+      AppSyncResolvers:
+        Mutation:
+          addPost:
+            Runtime:
+              Name: APPSYNC_JS
+              Version: 1.0.0
+            Functions:
+            - createPostLogInTable:
+                DataSource: PostsLogDataSource
+                InlineCode: |
+                  import { util } from '@aws-appsync/utils';
+
+                  export function request(ctx) {
+                    const id = util.autoId();
+                    const author = ctx.args.author
+                    const time = "1234567890"
+                    return dynamodbPutRequest({ key: {id}, values: {"author": author, "time": time}});
+                  }
+
+                  export function response(ctx) {
+                    return ctx.result;
+                  }
+
+                  /**
+                  * Helper function to create a new item
+                  * @returns a PutItem request
+                  */
+                  function dynamodbPutRequest({key, values}) {
```

Comment:
I agree that we should test `Functions` so I will add functions with `NONE` datasource for a more robust integration test.

Just to note, I won't be able to call it 2 times in this pipeline resolver because they are two separate datasources, `Posts` and `PostsLogs`. I will make 2 separate functions (which users probably will not do for something this simple), but still covers more functionality for the sake of testing.